gym_forex/envs/forex_env_v4.py

```diff
@@ -42,10 +41,8 @@ def __init__(self, **kwargs):
         self.episode_over=bool(0)
         # Number of past ticks per feature to be used as observations (1440min=1day, 10080=1Week, 43200=1month, )
         self.obs_ticks = kwargs['obsticks'] # best 48@ 700k
-        num_symbols = 1
```

Comment:
Just unused variables.

Reply:
Thanks for the changes, both thoughts for discussion are very convenient, but i don't have experience with both tools, so i will try to familiarize with them this chrismas, also i need to organize my projects better and their interactions,.

I think i will make a new agent using transformers (very excited to try) and finish the integration with data-logger before the refactoring of gym-forex and other tools, but suggestions are acceped about this or if there is something prioritary to work in before that.

The idea was that each tool can be used independently as feature-eng, but the size and number of the required projects increase, also their complexity and interactions, the more i approach the goal. If you can helpme to better plan the structure of all this, the workflow, and other things you mentioned like the best documentation, i would appreciate.

The goal is to be able to visualize performance, control (configure, deploy, pause, etc...), collaborate and use the result (evaluate trained models) of multiple NEAT optimization processes. 

The highest level project for this is the repo: singularity, an API that implements the migration operation in the NEAT algo by sharing best genomes with remote NEAT agents (with gym-fx for example) while they are training, for decentralized optimization of multiple models and also evaluation while optimization is occurring or have finished, but it is very outdated since many things should be done before working on it again, but i think it can be very useful, and more if its components can be used independently like gym-fx, data-logger or feature-eng.

Reply:
Thank you. I was considering integrating oandav20 into `preprocessing` so that `feature-engineering` and `feature-extractor` can easily do its magic to prep it for the environment. But while looking at `forex_env_v4.py`, I get confused on how the current data is formatted (what columns are which), since in `step()` the indicators are hardcoded to the column number.

I understand how this microservice architecture can increase overhead at first and be overwhelming, compared to a mono-repo. But it seems like it is a good route to go down since it provides modularity and independency. Which is what we want in a flexible RL environment to develop, test, and deploy.

I have worked with Docker a bit in the past and can attempt to dockerize this architecture once it's near completion, but more knowledge won't hurt! I am currently using oandav20 for all my forex data at the moment, so perhaps I can tackle this next if it is ok with you as you work on the transformer agent.

I think it'll be helpful and make things less daunting if we had a Miro Board to help brainstorm and draw everything out. Let me know if you want an invite to do so.

And happy holidays!

Reply:
@harveybc An overview of what you have going on so far in the Miro board, to my understanding. To reiterate this is a perfect architecture to turn into micro-services. But perhaps some suggestions so you don't have to reinvent the wheel so to speak:

1. [Dash by Plotly](https://dash.plotly.com/) and [Jupyter Notebooks](https://jupyter.org/) can do what the `visualizer` module can do. Idea is to quickly write and execute code in cells. Once finished, `.py` files can be made to run normally. There's even a VScode (text editor) plug-in for juptyer notebooks that have live variable explorer and tracker. This is what I currently use to be able to debug and run python code quickly.
2. [Mlflow](https://mlflow.org/docs/latest/index.html) and [airflow](https://airflow.apache.org/) can help deploy models to an API and retrieve and process data, respectively. And can be considered part or be the `preprocessor` and `feature-engineering` modules.

File organization can be something like:
```
.
├── README.md
├── scripts.sh
├── Storage
│   ├── airflow
│   ├── dash
│   ├── notebooks
│   ├── minio
│   │   └── storage
│   │       ├── airflow-files
│   │       │   ├── interested_tickers.xlsx
│   │       │   └── strategy.csv
│   │       ├── mlflow-models
├── docker-compose.yml
├── dockerfiles
│   ├── dockerfile_airflow
│   ├── dockerfile_dash
│   ├── dockerfile_jupyter_notebook
│   ├── dockerfile_minio
│   ├── dockerfile_mlflowserver
```
You can then spin up and teardown your entire architecture with just one command with the proper set up!

![Screen Shot 2022-12-06 at 7 11 12 PM](https://user-images.githubusercontent.com/6497318/206079073-40e66278-9844-4977-86a6-fd11bd598391.png)